jerry-core/vm/vm.c

```diff
@@ -2137,6 +2137,20 @@ vm_loop (vm_frame_ctx_t *frame_ctx_p) /**< frame context */
           *stack_top_p++ = ecma_copy_value (collection_p->buffer_p[tagged_idx]);
           continue;
         }
+        case VM_OC_PUSH_NEW_TARGET:
+        {
+          ecma_object_t *new_target_object = JERRY_CONTEXT (current_new_target);
+          if (new_target_object == NULL)
+          {
+            *stack_top_p++ = ECMA_VALUE_UNDEFINED;
+          }
+          else
+          {
+            ecma_ref_object (new_target_object);
```

Comment:
I'd assert that `JERRY_CONTEXT (current_new_target) != JERRY_CONTEXT_INVALID_NEW_TARGET `.

Reply:
Yeah, good idea. Will add it.